includes/class-wcpdf-settings-debug.php

```diff
@@ -36,120 +38,331 @@ public function debug_tools( $tab, $section ) {
+						<fieldset>
+							<div class="notice inline" style="display:none;"><p></p></div>
+						</fieldset>
+					</form>
+				</div>
+				<div class="tool">
+					<h4><span><?php _e( 'Import Settings', 'woocommerce-pdf-invoices-packing-slips' ); ?></span></h4>
+					<p><?php _e( 'Bring in the plugin settings from a .json file. This file can be obtained by downloading the settings on another site via the form above.', 'woocommerce-pdf-invoices-packing-slips' ); ?></p>
+					<form class="wpo_wcpdf_debug_tools_form" method="post" enctype="multipart/form-data">
+						<input type="hidden" name="debug_tool" value="import-settings">
+						<fieldset>
+							<input type="file" name="file" accept="application/json" required>
+							<a href="" class="button button-secondary submit"><?php _e( 'Import', 'woocommerce-pdf-invoices-packing-slips' ); ?></a>
+						</fieldset>
+						<fieldset>
+							<div class="notice inline" style="display:none;"><p></p></div>
+						</fieldset>
+					</form>
+				</div>
+			</div>
 		</div>
 		<br>
 		<?php
 	}
+	
+	public function ajax_debug_tools() {
+		check_ajax_referer( 'wpo_wcpdf_debug_nonce', 'nonce' );
+		
+		$data        = stripslashes_deep( $_REQUEST );
+		$debug_tools = [ 'export-settings', 'import-settings' ];
+		
+		if ( empty( $data['action'] ) || $data['action'] != 'wpo_wcpdf_debug_tools' ) {
+			return;
+		}
+		
+		if ( empty( $data['debug_tool'] ) || ! in_array( $data['debug_tool'], $debug_tools ) ) {
+			return;
+		}
+		
+		$debug_tool = esc_attr( $data['debug_tool'] );
+		
+		switch ( $debug_tool ) {
+			case 'export-settings':
+				$this->export_settings( $data );
+				break;
+			case 'import-settings':
+				$this->import_settings( $data );
+				break;
+		}
+		
+		wp_die();
+	}
+	
+	public function export_settings( $data ) {
+		extract( $data );
+		
+		if ( empty( $type ) ) {
+			$message = __( 'Export settings type is empty!', 'woocommerce-pdf-invoices-packing-slips' );
+			wcpdf_log_error( $message );
+			wp_send_json_error( compact( 'message' ) );
+		}
+		
+		$settings = [];
+		
+		switch ( $type ) {
+			case 'general':
+				$settings = WPO_WCPDF()->settings->general_settings;
+				break;
+			case 'debug':
+				$settings = WPO_WCPDF()->settings->debug_settings;
+				break;
+		}
+		
+		// maybe it's a document type settings request
+		if ( empty( $settings ) ) {
+			$documents = WPO_WCPDF()->documents->get_documents();
+			foreach ( $documents as $document ) {
+				if ( $type == $document->slug ) {
+					$settings = get_option( "wpo_wcpdf_documents_settings_{$document->get_type()}", [] );
+					break;
+				}
+			}
+			
+			if ( empty( $settings ) ) {
+				$message = __( 'Exported settings data is empty!', 'woocommerce-pdf-invoices-packing-slips' );
+				wcpdf_log_error( $message );
+				wp_send_json_error( compact( 'message' ) );
+			}
+		}
+		
+		$filename = apply_filters( 'wpo_wcpdf_export_settings_filename', sprintf( "{$type}-settings-export_%s.json", date( 'Y-m-d_H-i-s' ) ), $type );
```

Comment:
I'm using the slug here, not the title.

Reply:
I'm using the document slug.

Reply:
Maybe you can use the document label instead, sanitizing it for the file name? This way, the export file name will use the custom document title, if any.

Reply:
I understand but you are in fact importing settings for the default document, not the custom title, I think it makes the code confusing because I'm using the slug and in the middle changes to title. I think would be better to change the option title when exporting, eg: Proforma (Receipt)

Reply:
Done:

![Captura de ecrã de 2023-02-15 16-05-00](https://user-images.githubusercontent.com/533273/219083551-e2e10a2c-6688-4cfe-9fac-407b4e5c0ab9.png)